tests/internals.rs

```diff
@@ -639,12 +769,38 @@ fn test_invoke_tx() {
     // Extract invoke transaction fields for testing, as it is consumed when creating an account
     // transaction.
     let result = invoke_tx.execute(state, starknet_general_config).unwrap();
-
     let expected_execution_info = expected_transaction_execution_info();
 
     assert_eq!(result, expected_execution_info);
 }
 
+#[test]
+fn test_invoke_tx_state() {
+    let (starknet_general_config, state) = &mut create_account_tx_test_state().unwrap();
+    let expected_initial_state = expected_state_before_tx();
+    assert_eq!(state, &expected_initial_state);
+
+    let Address(test_contract_address) = TEST_CONTRACT_ADDRESS.clone();
+    let calldata = vec![
+        test_contract_address,                                       // CONTRACT_ADDRESS
+        Felt::from_bytes_be(&calculate_sn_keccak(b"return_result")), // CONTRACT FUNCTION SELECTOR
+        Felt::from(1),                                               // CONTRACT_CALLDATA LEN
+        Felt::from(2),                                               // CONTRACT_CALLDATA
+    ];
+    let invoke_tx = invoke_tx(calldata);
+
+    invoke_tx.execute(state, starknet_general_config).unwrap();
+
+    let expected_final_state = expected_state_after_tx();
+
+    assert_eq!(state.state_reader, expected_final_state.state_reader);
+    assert_eq!(state.cache, expected_final_state.cache);
+    assert_eq!(
+        state.contract_classes,
+        expected_final_state.contract_classes
+    );
```

Comment:
This can be reduced to a single assert with a dereference:
```suggestion
    assert_eq!(*state, expected_final_state);
```